File: plot/src/main/java/net/fwitz/math/plot/binary/split/color/SplitComplexColorFunction.java
package net.fwitz.math.plot.binary.split.color;

import net.fwitz.math.binary.split.SplitComplex;

import java.awt.*;
import java.util.function.BiFunction;

@FunctionalInterface
public interface SplitComplexColorFunction extends BiFunction<SplitComplex, SplitComplex, Color> {
    /**
     * Calculate the color for the given input and output values.
     *
     * @param c the input value
     * @param z the output value
     * @return the color to paint
     */
    @Override
    Color apply(SplitComplex c, SplitComplex z);
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/split/color/SplitComplexColorFunction.kt
package net.fwitz.math.plot.binary.split.color

import net.fwitz.math.binary.split.SplitComplex
import java.awt.Color

@FunctionalInterface
interface SplitComplexColorFunction : (SplitComplex, SplitComplex) -> Color {
    /**
     * Calculate the color for the given input and output values.
     *
     * @param c the input value
     * @param z the output value
     * @return the color to paint
     */
    override fun invoke(c: SplitComplex, z: SplitComplex): Color
}
